### ISSUE TITLE
Example binaries fail with "segmentation fault"

### ISSUE THREAD

**Issue by mihaiolteanu** [analogdevicesinc/libm2k#34]
![screenshot1570186014](https://user-images.githubusercontent.com/8273519/66202040-751bd600-e6ad-11e9-985b-99167032eb18.png)

Same for the `analog` and `digital` binaries.
Might be related to [#33](https://github.com/analogdevicesinc/libm2k/issues/33), as this, taken directly from the `examples/analog/getSamples.cpp` file, also fails with "segmentation fault",
```
int main(int argc, char* argv[]) {
    M2k *ctx = m2kOpen();
    ctx->calibrateADC();
    ctx->calibrateDAC();
}

**Comment by AlexandraTrifan** [comment]
Similar to #33 . I will return with some commits adding the missing checks in examples and in the libm2k documentation.

**Comment by AlexandraTrifan** [comment]
I added some checks for all the available examples in commit 8ab697b .
Also, getSamples was removed from the examples since it was only for testing purposes.
